backend/canisters/group_index/impl/src/updates/add_hot_group_exclusion.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{mutate_state, read_state, validate_user_is_super_admin, RuntimeState, ValidationError};
+use candid::Principal;
+use canister_tracing_macros::trace;
+use group_index_canister::add_hot_group_exclusion::{Response::*, *};
+use ic_cdk_macros::update;
+use types::{CanisterId, ChatId};
+
+#[update]
+#[trace]
+async fn add_hot_group_exclusion(args: Args) -> Response {
+    let PrepareResult {
+        caller,
+        user_index_canister_id,
+        ..
+    } = match read_state(|state| prepare(&args.chat_id, state)) {
+        Ok(ok) if !ok.is_excluded => ok,
+        Ok(_) => return ChatAlreadyExcluded,
+        Err(_) => return ChatNotFound,
+    };
+
+    match validate_user_is_super_admin(caller, user_index_canister_id).await {
+        Ok(_) => (),
+        Err(ValidationError::NotSuperAdmin) => return NotAuthorized,
+        Err(ValidationError::InternalError(error)) => return InternalError(error),
+    };
+
+    mutate_state(|state| commit(&args.chat_id, true, state));
+
+    Success
+}
+
+#[update]
+#[trace]
+async fn remove_hot_group_exclusion(args: Args) -> Response {
```

Comment:
This should return a different Response type

Reply:
See my comment above...